src/Model/LoginSession.php

```diff
@@ -175,4 +189,74 @@ public function getFriendlyUserAgent()
 
         return sprintf('%s on %s', $result->ua->family, $result->os->toString());
     }
+
+    /**
+     * @param Member|null $member
+     * @param HTTPRequest|null $request
+     * @return LoginSession|null
+     */
+    public static function getCurrentLoginSession(Member $member = null, HTTPRequest $request = null)
```

Comment:
Not keep on having public static functions unless there's a proper need for it, though looking at the contents of the function it looks as though it was copied from somewhere else.  I don't have my head fully around where this code is used.

Given this PR is primarily a JS ui PR, and this is still an unreleased module, I'm thinking we go light on reviewing the PHP stuff for now with the intention of a follow up PR to look at tidying up the backend

Reply:
Why should we not do this?

Reply:
static public methods like this are sometimes just code smell which leads to tightly coupled code.  I don't know enough about the session manager module and intended usage of this method to really determine whether it should be static or not though, and there's already so much going on in this PR on the client side I'd rather just focus on that for now